dex/src/main/scala/com/wavesplatform/dex/api/websockets/actors/WsHandlerActor.scala

```diff
@@ -88,40 +111,38 @@ object WsHandlerActor {
                   // TODO DEX-700 test for order book that hasn't been created before
                   if (subscribe.depth <= 0) clientRef ! WsError.from(error.RequestArgumentInvalid("depth"), time.getTimestamp())
                   else
-                    assetPairBuilder.validateAssetPair(subscribe.key).value.onComplete {
-                      case Success(Left(e)) => clientRef ! WsError.from(e, time.getTimestamp())
-                      case Success(Right(_)) =>
-                        matcherRef ! MatcherActor.AggregatedOrderBookEnvelope(
-                          subscribe.key,
-                          AggregatedOrderBookActor.Command.AddWsSubscription(clientRef)
-                        )
-                      case Failure(e) =>
-                        context.log.warn(s"An error during validation the asset pair ${subscribe.key}", e)
-                        clientRef ! WsError.from(error.WavesNodeConnectionBroken, time.getTimestamp())
-                    }
+                    context.pipeToSelf(assetPairBuilder.validateAssetPair(subscribe.key).value) { ProcessAssetPairValidationResult(subscribe.key, _) }
```

Comment:
Suggest a small modification. We could validate an asset pair and send a message this actor about a successful result only. This causes:
1. We get a smaller pressure on WsHandlerActor mailbox
2. ProcessAssetPairValidationResult is simpler, e.g.:
    ```scala
    private[WsHandlerActor] case class AssetPairValidated(assetPair: AssetPair) extends Command
    ```

Reply:
What we will return in case of failure? We have to transform `Try[Either[MatcherError, AssetPair]]` to `Message`

Reply:
I mean we could do something like:
```
assetPairBuilder.validateAssetPair(subscribe.key).value.onComplete {
   case Success(Left(e))  => clientRef ! WsError.from(e, time.getTimestamp())
   case Success(Right(_)) => context.self ! AssetPairValidated(subscribe.key)
   case Failure(e) =>
     context.log.warn(s"An error during validation the asset pair ${subscribe.key}", e)
     clientRef ! WsError.from(error.WavesNodeConnectionBroken, time.getTimestamp())
 }
```

Reply:
Done